### PR TITLE
fix(run): no-cache flag

### PR DESCRIPTION
**Pull request by Xenfo**
Fixed the `—no-cache` flag.

### CODE REVIEW

**Review by jaredpalmer — CHANGES_REQUESTED** [review]
Can you add e2e test for this?

**Comment by Xenfo** [comment]
I guess I'm a bit mixed up about what `--no-cache` does thought it was like `--force`. Can you lmk so I can add E2E.

**Comment by jaredpalmer** [comment]
- --force ignores a cache hit and always executes 
- --no-cache avoids saving to cache (calling .Put)

**Comment by jaredpalmer** [comment]
Needs e2e tests, but looks fine

**Comment by Xenfo** [comment]
@jaredpalmer added E2E tests

File: cli/scripts/e2e/e2e.ts [hunk 1]
```diff
@@ -85,15 +85,46 @@ function runSmokeTests<T>(
...
+        "Packages in scope"
+      );
+      assert.fixture(
+        `• Running test in 1 packages`,
+        sinceCommandOutputNoCache[2],
+        "Runs only in changed packages"
+      );
+      assert.fixture(
+        sinceCommandOutputNoCache[3],
+        `a:test: cache miss, executing ${getHashFromOutput(
```
Comment (jaredpalmer) [inline cli/scripts/e2e/e2e.ts:121]: This isn’t fully testing —no-cache. Need to assert that the outputted log file was not copied into the cache folder

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: cli/scripts/e2e/e2e.ts (hunk 1 above)
Comment (Xenfo) [reply cli/scripts/e2e/e2e.ts:121]: Isn't that assumed if https://github.com/vercel/turborepo/blob/d81311665f337788ba9459f5173c671f12214cb2/cli/scripts/e2e/e2e.ts#L128 is a cache miss?

Comment (jaredpalmer) [reply cli/scripts/e2e/e2e.ts:121]: ahh true that. :smart:

File: cli/scripts/e2e/e2e.ts [hunk 2]
```diff
@@ -63,9 +63,9 @@ function runSmokeTests<T>(
   });
 
   suite(
-    `${npmClient} runs tests and logs ${
+    `${npmClient} runs tests and logs${
```
Comment (jaredpalmer) [inline cli/scripts/e2e/e2e.ts:66]: add back space here
```suggestion
    `${npmClient} runs tests and logs ${
```

File: cli/scripts/e2e/e2e.ts [hunk 3]
```diff
@@ -85,15 +85,46 @@ function runSmokeTests<T>(
   );
 
   suite(
-    `${npmClient} handles filesystem changes ${
+    `${npmClient} handles filesystem changes${
```
Comment (jaredpalmer) [inline cli/scripts/e2e/e2e.ts:88]:
```suggestion
    `${npmClient} handles filesystem changes ${
```

File: cli/scripts/e2e/e2e.ts [hunk 4]
```diff
@@ -211,9 +242,7 @@ function runSmokeTests<T>(
     // Test `turbo prune --scope=a`
     // @todo refactor with other package managers
     suite(
-      `${npmClient} + turbo prune ${
-        options.cwd ? " from " + options.cwd : ""
-      } `,
+      `${npmClient} + turbo prune${options.cwd ? " from " + options.cwd : ""}`,
```
Comment (jaredpalmer) [inline cli/scripts/e2e/e2e.ts:245]:
```suggestion
      `${npmClient} + turbo prune ${options.cwd ? " from " + options.cwd : ""}`,
```

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**